tests/rptest/tests/self_test_test.py

```diff
@@ -52,6 +56,52 @@ def test_self_test(self):
                 assert 'warning' not in report
                 assert report['duration'] > 0, report['duration']
 
+        num_nodes = 3
+
+        # Ensure the results appear as expected. Assertions aren't performed
+        # on specific results, but rather what tests are oberved to have run
+        reports = flat_map(lambda node: node['results'], node_reports)
+
+        # Ensure 4 disk tests per node, read/write & latency/throughput
+        disk_results = [r for r in reports if r['test_type'] == 'disk']
+        expected_disk_results = num_nodes * 4
+        assert len(
+            disk_results
+        ) == expected_disk_results, f"Expected {expected_disk_results} disk reports observed {len(disk_results)}"
+
+        # Assert properties of the network results hold true
+        network_results = [r for r in reports if r['test_type'] == 'network']
+
+        # Ensure no other result sets exist
+        assert len(disk_results) + len(network_results) == len(reports)
+
+        # Ensure nCr network test reports, clusterwide
+        assert len(network_results) == comb(
+            num_nodes, 2
+        ), f"Expected {comb(num_nodes,2)} reports observed {len(network_results)}"
+
+        # Assert correct combinations of nodes were chosen
+        def seperate_pairings(result):
```

Comment:
nit: for anyone else that has to visit this test, might be worth explaining the intent, like "The network self tests take care to ensure only one test runs between any given pair of nodes. Check that that is the case." or something